### DISCUSSION TITLE
How to extract data from `Term::Literal`?

### DISCUSSION THREAD

**Discussion by hoijui** [oxigraph/oxigraph#829] [Q&A]
I have `Term` which is of the variant `Term::Literal`.
More specifically this means it is of type `Term::Literal(Literal(LiteralContent))`.

with:

```
enum LiteralContent {
    String(String),
    LanguageTaggedString { value: String, language: String },
    TypedLiteral { value: String, datatype: NamedNode },
}
```

the 2nd/middle `Literal` in `Term::Literal(Literal(LiteralContent))` is private though (and so is `LiteralContent` within it), and I see no functions to get to it. Thus the only way to get to the actual data, is to `.display()`/`.fmt()`/`.value()` the `Term::Literal` ...?
(It seems wrong and inefficient to me)

**Comment by Tpt** [comment]
Hey! Thank you for trying Oxigraph. There are a few accessors methods: `value`, `language`, `datatype` for references and [`destruct`](https://docs.rs/oxigraph/latest/oxigraph/model/struct.Literal.html#method.destruct) if you want to own the underlying structs
